Go Feature Guide: Editing (#2904)
temporalio/documentation: docs/develop/go/core-application.mdx

description: The Foundations section of the Temporal Developer's guide introduces essential concepts for building and running a Temporal Application, outlining steps to start Workflow and Activity Execution, running development Workers, and installing the Temporal CLI.
---

This page shows how to do the following:

> In this section you can find the following:

Reviewer: Do you need both 89 and 91?

In this section you can find the following:

- [Run a development Temporal Service](#run-a-development-server)
- [Develop a Workflow](#develop-workflows)
- [Develop an Activity](#activity-definition)
- [Start an Activity Execution](#activity-execution)
- [Run a dev Worker](#develop-worker)
- [Run a Temporal Cloud Worker](#run-a-temporal-cloud-worker)

## How to install the Temporal CLI and run a development server {#run-a-development-server}
## Develop a basic Workflow {#develop-workflows}

> This section describes how to install the [Temporal CLI](/cli) and run a development Temporal Service.

Reviewer: This is HUGE removal of material. Just making sure it's intentional.

Author: Haven't looked back at this but any removals are likely because they are being moved to a new spot to align with the migration. The .NET one was a manual migration since the rest weren't, need to make sure they are still manually moved. ✔️

}
```

> ### Customize Workflow Type in Go {#customize-workflow-type}

Reviewer 2: You replaced "in Go" with "using the Go SDK" above so maybe you can replace this with 'using the Go SDK' as well

Reviewer: +1 to Angela, who got there before me. 

Also maybe Types instead of Type

- `workflow.Context` This is a replacement for `context.Context`.
  See [Tracing](/develop/go/observability#tracing-and-context-propogation) for more information about context propagation.

> ## Develop an Activity Definition in Go {#activity-definition}

Reviewer 2: You replaced "in Go" with "using the Go SDK" above so maybe you can replace this with 'using the Go SDK' as well

}
```

> ### Customize Activity Type in Go {#customize-activity-type}

Reviewer 2: You replaced "in Go" with "using the Go SDK" above so maybe you can replace this with 'using the Go SDK' as well

Reviewer: +1 also Type/Types

### Go ActivityOptions reference {#activity-options-reference}

> **How to use Go ActivityOptions reference using the Go SDK**

Reviewer 2: ```suggestion
**How to configure `ActivityOptions` using the Go SDK**
```

}
```

> ### Get the results of an Activity Execution {#get-activity-results}

Reviewer: Consider 'retrieve' vs 'get'

Reviewer: Also 'Activity Execution results' is shorter and more direct than 'of an'

> **How to get the results of an Activity Execution using the Go SDK**

Reviewer: ditto.

// ...
```

> ### Set WorkerOptions in Go {#workeroptions}

Reviewer: Backtick?

> **How to set WorkerOptions using the Go SDK**

Reviewer 2: ```suggestion
**How to set 1WorkerOptions` using the Go SDK**
```

Same for line 894

Reviewer: Except without the number 1

Reviewer 2: Whooops :)

w.RegisterWorkflow(WorkflowC)
```

### Set RegisterWorkflowOptions in Go {#registerworkflowoptions}

> **How to set RegisterWorkflowOptions using the Go SDK**

Reviewer 2: ```suggestion
**How to set `RegisterWorkflowOptions` using the Go SDK**
```
Same for line 1433

// ...
```

### Set RegisterActivityOptions in Go {#registeractivityoptions}

> **How to set RegisterActivityOptions using the Go SDK**

Reviewer: Backtick the code face or reword
